src/basic/sysctl-util.c

```diff
@@ -44,8 +44,37 @@ char* sysctl_normalize(char *s) {
         return s;
 }
 
-int sysctl_write(const char *property, const char *value) {
+static int shadow_update(Hashmap **shadow, const char *property, const char *value) {
+        _cleanup_free_ char *k = NULL, *v = NULL, *cur_k = NULL, *cur_v = NULL;
+        int r;
+
+        assert(shadow);
```

Comment:
it might be nice to simply make this a NOP if `shadow` is NULL. i.e. replace this line with `if (!shadow) return 0;` or so.

Reply:
Before the property and value assert, so it's a NOP even if other arguments are NULL?

Reply:
nope, after